src/HeapOrdering.sol

```diff
@@ -72,68 +72,55 @@ library HeapOrdering {
     /// @dev The heap may lose its invariant about the order of the values stored.
     /// @dev Only call this function with an index within array's bounds.
     /// @param _heap The heap to modify.
-    /// @param _index The index of the account in the heap to be set.
     /// @param _account The account to set the `_index` to.
+    /// @param _index The index of the account in the heap to be set.
     function setAccount(
         HeapArray storage _heap,
-        uint256 _index,
-        Account memory _account
+        Account memory _account,
+        uint256 _index
     ) private {
         _heap.accounts[_index] = _account;
         _heap.indexOf[_account.id] = _index;
     }
 
-    /// @notice Swaps two accounts in the `_heap`.
-    /// @dev The heap may lose its invariant about the order of the values stored.
-    /// @dev Only call this function with indexOf within array's bounds.
-    /// @param _heap The heap to modify.
-    /// @param _index1 The index of the first account in the heap.
-    /// @param _index2 The index of the second account in the heap.
-    function swap(
-        HeapArray storage _heap,
-        uint256 _index1,
-        uint256 _index2
-    ) private {
-        if (_index1 == _index2) return;
-        Account memory accountOldIndex1 = _heap.accounts[_index1];
-        Account memory accountOldIndex2 = _heap.accounts[_index2];
-        setAccount(_heap, _index1, accountOldIndex2);
-        setAccount(_heap, _index2, accountOldIndex1);
-    }
-
     /// @notice Moves an account up the heap until its value is smaller than the one of its parent.
-    /// @dev This functions restores the invariant about the order of the values stored when the account at `_index` is the only one with value greater than what it should be.
+    /// @dev This functions restores the invariant about the order of the values stored when the account to move is the only one with value greater than what it should be.
     /// @param _heap The heap to modify.
+    /// @param _accountToShift The account to move.
     /// @param _index The index of the account to move.
-    function shiftUp(HeapArray storage _heap, uint256 _index) private {
-        Account memory accountToShift = _heap.accounts[_index];
-        uint256 valueToShift = accountToShift.value;
+    function shiftUp(
+        HeapArray storage _heap,
+        Account memory _accountToShift,
+        uint256 _index
+    ) private {
         Account memory parentAccount;
+        uint256 parentIndex;
```

Comment:
You're right, almost all the tests slighty favor not creating the variable. But are we sure that it's not because all the tests are done with a little number of users?

As I see it, it's the same as with [Quentin's remark](https://github.com/morpho-dao/morpho-data-structures/pull/113#discussion_r1172367111), there is a higher initialization cost but a lower cost at each iteration.

Also, the only test that is in favor of creating the variable is one that does a lot of iterations in `shiftUp` (snapshot with variable, diff without variable):
![image](https://user-images.githubusercontent.com/108467407/233640026-83dde4ca-846c-4cf2-a010-b87b387bbe83.png)

So I wrote the following test that basically does the most iterations possible in `shiftUp`:
```solidity
function testInsertLot() public {
        MAX_SORTED_USERS = type(uint256).max;
        for (uint160 i = 1; i < 10000; i++) update(address(i), 0, i);
    }
```
And the result is the following (snapshot with variable, diff without variable): 
![image](https://user-images.githubusercontent.com/108467407/233642079-17f40567-39ca-4c2f-87ae-3b6185395430.png)

I took the opportunity to test with [Quentin's remark](https://github.com/morpho-dao/morpho-data-structures/pull/113#discussion_r1172367111), and the conclusion is the same (snapshot without variable, diff with variable):
![image](https://user-images.githubusercontent.com/108467407/233642663-c6f5ae7f-5113-48a5-bf57-268ed5412a7f.png)

I think this highlights an overall issue with the testing suite; there is no test that performs huge operations on the heap.

Reply:
@QGarchery it might interest you as well

Reply:
Interesting ! In practice [maxSortedUsers = 8](https://etherscan.io/address/0x777777c9898d384f785ee44acfe945efdff5f3e0#readProxyContract#F27) though. As you want in the end, no blocker.

Reply:
If we can have an estimate on the breakeven point that would be ideal. Let's say it's 4 iterations, then maybe it's not worth it to add the variable: remember that the deployed heap is very small, and it should be the same for anyone deploying on mainnet at the moment. Otherwise, I'd be in favor of having the variable both times, for readability and because the network might scale such that bigger heaps are possible

Reply:
according to my tests:
- creating the variable `parentIndex` is better when there is at least 1 iteration
- creating the variable `valueToShift` is better when there are at least 2 iterations
 So i'd go for creating both variables, even if `valueToShift` is not always beneficial in Morpho because the heap has a height of 2. This repo would be a little bit less specific to Morpho.